bagit.py

```diff
@@ -237,14 +254,31 @@ def tagmanifest_files(self):
                 yield f
 
     def compare_manifests_with_fs(self):
-        files_on_fs = set(self.payload_files())
-        files_in_manifest = set(self.payload_entries().keys())
+        """
+        Compare the filenames in the manifests to the filenames present on the
+        local filesystem and returns two lists of the files which are only
+        present in the manifests and the files which are only present on the
+        local filesystem, respectively.
+        """
+
+        # We compare the filenames after Unicode normalization so we can
+        # reliably detect normalization changes after bag creation:
+        files_on_fs = set(normalize_unicode(i) for i in self.payload_files())
+        files_in_manifest = set(normalize_unicode(i) for i in self.payload_entries().keys())
 
         if self.version == "0.97":
-            files_in_manifest = files_in_manifest | set(self.missing_optional_tagfiles())
+            files_in_manifest.update(self.missing_optional_tagfiles())
+
+        only_on_fs = list()
+        only_in_manifest = list()
 
-        return (list(files_in_manifest - files_on_fs),
-                list(files_on_fs - files_in_manifest))
+        for i in files_on_fs.difference(files_in_manifest):
+            only_on_fs.append(self.normalized_filesystem_names[i])
```

Comment:
`i` has been passed through `normalize_unicode` at this point so we can't assume that it will still be the same byte values as stored in the filesystem.